### ISSUE TITLE
Rule request: newValue ignored on var set

### ISSUE THREAD

**Issue by fpg1503** [realm/SwiftLint#1136]
I've seen a lot of times people using the variable name instead of `newValue` which leads to bugs:

```swift
var aValue: String {
    get {
        return Persister.shared.aValue
    }
    set {
        Perister.shared.aValue = aValue //should be newValue
    }
}
```

**Comment by marcelofabri** [comment]
A small detail here is that you can use another name for the parameter. For example:

```swift
// shouldn't trigger
var aValue: String {
    get {
        return Persister.shared.aValue
    }
    set(persistedValue) {
        Perister.shared.aValue = persistedValue
    }
}
```

**Comment by marcelofabri** [comment]
There's an open ticket on Swift JIRA for this as well: [SR-964](https://bugs.swift.org/browse/SR-964)

**Comment by marcelofabri** [comment]
Closing this as https://github.com/apple/swift/pull/11465 was merged and it'll be a compiler warning in future Swift versions 🎉

**Comment by ZevEisenberg** [comment]
Note that a case was missed in the resolution to [SR-964](
https://bugs.swift.org/browse/SR-96), and the [PR for the fix](https://github.com/apple/swift/pull/14256) never ended up getting merged, so SwiftLint could still provide a band-aid (or anyone who's interested could pick up that PR and port it to Swift 4.2 or 5.0).
